File: app-android/src/main/java/com/ustadmobile/port/android/view/RangeSeekCustom.java
package com.ustadmobile.port.android.view;

import android.content.Context;
import android.util.AttributeSet;

import com.crystal.crystalrangeseekbar.widgets.CrystalRangeSeekbar;

public class RangeSeekCustom extends CrystalRangeSeekbar {
    private final int THUMB_SIZE = 40;
    public RangeSeekCustom(Context context) {
        super(context);
    }

    public RangeSeekCustom(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public RangeSeekCustom(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected float getBarHeight() {
        return 5;
    }

    @Override
    protected float getThumbWidth() {
        return THUMB_SIZE;
    }

    @Override
    protected float getThumbHeight() {
        return THUMB_SIZE;
    }
}

File: app-android/src/main/kotlin/com/ustadmobile/port/android/view/RangeSeekCustom.kt
package com.ustadmobile.port.android.view

import android.content.Context
import android.util.AttributeSet

import com.crystal.crystalrangeseekbar.widgets.CrystalRangeSeekbar

class RangeSeekCustom : CrystalRangeSeekbar {
    private val THUMB_SIZE = 40

    override fun getBarHeight(): Float {
        return 5f
    }

    override fun getThumbWidth(): Float {
        return THUMB_SIZE.toFloat()
    }

    override fun getThumbHeight(): Float {
        return THUMB_SIZE.toFloat()
    }

    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int)
            : super(context, attrs, defStyleAttr) {}
}
